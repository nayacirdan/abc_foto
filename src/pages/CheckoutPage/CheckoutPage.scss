.head-title {
  font-size: 26px;
  text-transform: uppercase;
  display: flex;
  margin-top: 35px;
}
.order-wrapper {
  padding-top: 35px;
  width: 100%;
  max-width: 1140px;
  margin: 0 auto;
}

.order-container {
  width: 490px;
}

.block-title {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: baseline;

  .small-text {
    font-weight: 200;
    font-size: 12px;
    display: flex;
    margin-bottom: 25px;
    margin-top: 35px;
  }
}

.big-text {
  font-size: 20px;
  display: flex;
  margin-bottom: 25px;
  margin-top: 35px;
}

.order-delivery {
  display: flex;
  flex-direction: column;
}

.MuiSvgIcon-root {
  color: #51AD33;
}

.radio-group label .MuiTypography-body1 {
  font-size: 14px ;
}

.MuiTypography-body1 {
  font-size: 0.88rem;
}

.comments {
  width: 468px;
  border: 1px solid #D3D7DA;
  border-radius: 3px;
  padding: 10px 10px 100px 10px;
  margin: 30px 0;
}

.checkout-container {
  right: 10%;
  background: #F9F9FB;
  padding: 50px 70px;

  .MuiButtonBase-root.MuiButton-root.MuiButton-contained {
    width: 300px;
    background: #51AD33;
    color: white;
    text-transform: uppercase;
    display: flex;
    margin-top: 30px;
    padding: 10px 0;
  }
}

.checkout-group {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

.checkout-title {
  width: 370px;
  margin: 0;
  padding: 10px 60px;
  background-color: #98A1AF;
  color: white;
}

.MuiSelect-outlined.MuiSelect-outlined{
  display: flex;
}

.checkout-item {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  margin-bottom: 30px;

  &-image {
    border-style: none;
    width: 80px;
    height: 80px;
    padding: 10px;
    background: #F7F5F6;
  }

  &-price {
    font-size: 16px;
  }

  .item-description {
    display: flex;
    flex-direction: column;
    align-items: flex-start;

    &-title {
      font-size: 13px;
    }

    &-code {
      font-size: 11px;
    }
  }
}

.order-sum {
  font-size: 26px;
  font-weight: bold;
  margin-top: 100px;
  display: flex;
}
