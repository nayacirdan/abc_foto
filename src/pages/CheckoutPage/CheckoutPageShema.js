// import * as yup from 'yup';

// const phone_regex = /^\+?3?8?(0\d{9})$/;
// const email_regex = !/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i;
// const no_spacing_regex = /^[a-zA-Z0-9]*$/;
// const only_letters_regex = /^[a-zA-Z]+$/;

// export const CheckoutSchema = yup.object().shape({
//   name: yup.string()
//     .min(2, 'Min name length is 2 symbols')
//     .matches(only_letters_regex, 'Name can contain only letters')
//     .matches(no_spacing_regex, 'Enter your name without spacing')
//     .required('This field is required'),
//   email: yup.string()
//     .matches(email_regex, 'Enter correctly Email')
//     .required('his field is required'),
//   phone: yup.string()
//     .matches(phone_regex, 'Your phone number should look like +380938107132')
//     .min(13)
//     .max(13)
// });
