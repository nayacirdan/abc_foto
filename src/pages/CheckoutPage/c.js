export const productsNotJson = [{
  enabled: true,
  imageUrls: ['https://res.cloudinary.com/miratsiupiak/image/upload/v1592831421/products/canon_2/image_2_uur7dj.jpg', 'https://res.cloudinary.com/miratsiupiak/image/upload/v1592831421/products/canon_2/image_3_n53uc9.jpg', 'https://res.cloudinary.com/miratsiupiak/image/upload/v1592831420/products/canon_2/image_1_wh8z5q.jpg', 'https://res.cloudinary.com/miratsiupiak/image/upload/v1592831421/products/canon_2/image_5_a6gzhf.jpg'],
  quantity: 1,
  _id: '5ee0e3835ecb8a365496c723',
  name: 'canon eos 6d mark ii body',
  currentPrice: 32999,
  previousPrice: 34699,
  categories: 'photocameras',
  productUrl: '/photocameras',
  brand: 'Canon',
  article: 54895152,
  isAvailable: true,
  description: 'Матрица CMOS 35.9x24 мм, 26.2 Мп / поддержка карт памяти SD/SDHC/SDXC / LCD-дисплей 3" / Full HD-видео / питание от литий-ионного аккумулятора / 144 x 110.5 x 74.8 мм, 765 г (включая аккумулятор и карту памяти)',
  type: 'зеркальные',
  matrix: 'CMOS',
  isHit: true,
  itemNo: '545096',
  date: '2020-06-10T13:43:31.997Z',
  __v: 0,
  withLens: false,
  color: 'черный',
  cameraMatrix: ['35.9 x 24 мм CMOS', 'Количество эффективных пикселей: приблизительно 26.2 мегапикселя', 'Общее количество пикселей: приблизительно 27.1 мегапикселя', 'Соотношение сторон: 3:2', 'Фильтр нижних частот: встроенный/фиксированный', 'Очистка датчика изображения: встроенная система очистки EOS', 'Тип цветного светофильтра: основные цвета'],
  classPro: 'Professional',
  interfaceProp: ['GPS', 'Micro USB', 'NFC', 'WI-Fi'],
  photosensitivity: ['Фото: авто (100 - 40000), 100 - 40000 (с шагом 1/3 ступени или целая ступень), в ручном режиме чувствительность ISO может быть расширена до L: 50, H1: 51200, H2: 102400', 'Видео: авто (100 - 25600), 100 - 25600 (с шагом 1/3 ступени или целая ступень), в автоматическом и ручном режиме чувствительность ISO может быть расширена до H1: 51200. H2: 102400'],
  specialFeatures: ['NFC (соответствие NFC Forum Type 3/4 Tag (динамическое))'],
  videoRecording: 'Full HD (1920x1080)',
  guarantee: '24 месяца',
  weight: 'Камера - 765 г (включая аккумулятор и карту памяти)',
  label: 'canon eos 6d mark ii body',
  value: 'canon eos 6d mark ii body'
}, {
  enabled: true,
  imageUrls: ['https://res.cloudinary.com/miratsiupiak/image/upload/v1592831011/products/canon_1/image_2_bgriub.jpg', 'https://res.cloudinary.com/miratsiupiak/image/upload/v1592831011/products/canon_1/image_3_v2gjof.jpg', 'https://res.cloudinary.com/miratsiupiak/image/upload/v1592831011/products/canon_1/image_4_vpylfh.jpg', 'https://res.cloudinary.com/miratsiupiak/image/upload/v1592831004/products/canon_1/image_1_kw8g3l.jpg'],
  quantity: 1,
  _id: '5ee0e2115ecb8a365496c722',
  name: 'canon eos 800d body black',
  currentPrice: 15899,
  categories: 'photocameras',
  productUrl: '/photocameras',
  brand: 'Canon',
  article: 16875476,
  isAvailable: true,
  description: 'Матрица 22.3 x 14.9 мм, 24.2 Мп / поддержка карт памяти SD/SDHC/SDXC / Сенсорный ЖК-дисплей с переменным углом наклона Clear View II TFT 3" / FullHD-видео / питание от литий-ионного аккумулятора / 131 x 76.2 x 99.9 мм, 532 г',
  type: 'зеркальные',
  matrix: 'Полноразмерная',
  isHit: true,
  itemNo: '201312',
  date: '2020-06-10T13:37:21.739Z',
  __v: 0,
  withLens: false,
  color: 'черный',
  cameraMatrix: ['22.3 x 14.9 мм CMOS', 'Эффективные пиксели: Прибл. 24.2 мегапикселя', 'Общее количество пикселей: Прибл. 25.8 мегапикселя', 'Соотношение сторон: 3:2'],
  classPro: 'Professional',
  exposureMode: ['Автоматический режим: доступен в режиме покадровой автофокусировки при оценочном замере по достижении фокусировки', 'Ручная: с помощью кнопки фиксации AE в режимах творческой зоны'],
  lens: ['Совместимость со всеми объективами EF, EF-S'],
  photosensitivity: ['Авто (100-25600), 100-25600 (при шаге 1/3 ступени или целая ступень)', 'Доступно расширение ISO до H: 51200', 'Во время съемки видео: Авто (100–12800), 100–12800 (при шаге 1/3 ступени или целая ступень)', 'чувствительность ISO может быть расширена до H: 25600'],
  videoRecording: 'Full HD (1920x1080)',
  interfaceProp: ['GPS', 'Micro USB', 'NFC', 'WI-Fi'],
  guarantee: '24 месяца официальной гарантии от производителя',
  specialFeatures: ['NFC'],
  weight: '532 г (включая аккумулятор и карту памяти)',
  catalogs: 'digital-techniques',
  label: 'canon eos 800d body black',
  value: 'canon eos 800d body black'
}, {
  enabled: true,
  imageUrls: ['https://res.cloudinary.com/miratsiupiak/image/upload/v1592831011/products/canon_1/image_2_bgriub.jpg', 'https://res.cloudinary.com/miratsiupiak/image/upload/v1592831011/products/canon_1/image_3_v2gjof.jpg', 'https://res.cloudinary.com/miratsiupiak/image/upload/v1592831011/products/canon_1/image_4_vpylfh.jpg', 'https://res.cloudinary.com/miratsiupiak/image/upload/v1592831004/products/canon_1/image_1_kw8g3l.jpg'],
  quantity: 1,
  _id: '5ee0e2115ecb8a365496c722',
  name: 'canon eos 800d'
}
];

/* const values = {
  email: 'ksusha.cirdan@gmail.com',
  phone: '0937699607',
  comment: 'апвапвап',
  deliveryMethod: 'self',
  paymentMethod: 'cash'
}; */
/*
const sendData = (values) => {

};
const NAC_data = {
  email: values.email,
  mobile: values.phone,
  comment: values.comment,
  deliveryMethod: values.deliveryMethod,
  paymentMethod: values.paymentMethod,
  letterSubject: 'Thank you for order! You are welcome!',
  letterHtml: '<h1>Your order is placed. OrderNo is 023689452.</h1><p>{Other details about order in your HTML}</p>',
  canceled: false,
  products: JSON.stringify(productsNotJson)
};

const AC_data = {
  email: values.email,
  mobile: values.phone,
  comment: values.comment,
  deliveryMethod: values.deliveryMethod,
  paymentMethod: values.paymentMethod,
  letterSubject: 'Thank you for order! You are welcome!',
  letterHtml: '<h1>Your order is placed. OrderNo is 023689452.</h1><p>{Other details about order in your HTML}</p>',
  canceled: false,
  products: []
};
*/
