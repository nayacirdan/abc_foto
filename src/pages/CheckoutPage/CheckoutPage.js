import React from 'react';
import {Formik, Form, Field} from 'formik';
import SimpleModal from '../../components/Modals/ConfirmOrderModal';
import TextField from '@material-ui/core/TextField';
import './CheckoutPage.scss';
import {toggleOrderModal} from '../../store/actions/actions';
import {useDispatch, useSelector} from 'react-redux';
import PersonalInfo from './PersonalInfo/PersonalInfo';
import DeliverMethods from './DeliverMethods/DeliverMethods';
import PaymentMethods from './PaymentMethods/PaymentMethods';
import {Button} from '@material-ui/core';
import axios from 'axios';
import {productsNotJson} from './c';
import ItemsInCart from './ItemsInCart/ItemsInCart';

function CheckoutPage () {
  const dispatch = useDispatch();
  const customer = useSelector(state => state.getCustomer.customerInfo);
  const productsInCart = useSelector(state => state.cartReducer.cartInfo);
  console.log(productsInCart);
  console.log(customer._id);
  let customerName;

  if (customer.firstName || customer.lastName) {
    customerName = `${customer.firstName} ${customer.lastName}`;
  }

  /* const getCartProducts = () => {
      return new Promise((resolve, reject) => {
        setTimeout(() => resolve(mockCart), 1000);
      });
    }; */

  /* useEffect(() => {
      getCartProducts().then(result => {
        setProducts(result);
      });
    }, []); */

  const NAC_data = {
    email: 'ksusha.cirdan@gmail.com',
    mobile: '0937699607',
    letterSubject: 'Thank you for order! You are welcome!',
    letterHtml: '<h1>Your order is placed. OrderNo is 023689452.</h1><p>{Other details about order in your HTML}</p>',
    canceled: false,
    products: JSON.stringify(productsNotJson)
  };

  const AC_data = {
    customerId: customer._id || '5f1318029ba9f42208815d10',
    email: 'ksusha.cirdan@gmail.com',
    mobile: '0937699607',
    letterSubject: 'Thank you for order! You are welcome!',
    letterHtml: '<h1>Your order is placed. OrderNo is 023689452.</h1><p>{Other details about order in your HTML}</p>',
    canceled: false,
    products: []
  };

  console.log(JSON.stringify(NAC_data));
  console.log(JSON.stringify(AC_data));

  const sendOrderNAC = () => {
    axios.post('/orders', NAC_data)
      .then(newOrder => {
        console.log('NAC', newOrder.data);
        /* Do something with newOrder */
      })
      .catch(err => {
        console.log('NAC', err, err.message);
        /* Do something with error, e.g. show error to user */
      });
  };

  const sendOrderAC = () => {
    axios.post('/orders', AC_data)
      .then(newOrder => {
        console.log('AC', newOrder.data);
        /* Do something with newOrder */
      })
      .catch(err => {
        console.log('AC', err, err.message);
        /* Do something with error, e.g. show error to user */
      });
  };

  const handleSubmit = (values) => {
    dispatch(toggleOrderModal(true));
    console.log('values=', values);
  };

  const validate = values => {
    const errors = {};

    if (!values.email) {
      errors.email = 'Required';
    } else if (
      !/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i.test(values.email)
    ) {
      errors.email = 'Invalid email address';
    }

    if (!values.phone) {
      errors.phone = 'Required';
    } else if (!/^\d{10}$/i.test(values.phone) || values.phone.length !== 10) {
      errors.phone = 'Invalid phone number';
    }

    if (!values.name) errors.name = 'Required';

    return errors;
  };

  return (
    <div className="order-wrapper">
      <h2 className="head-title">Оформление заказа</h2>
      <Formik
        initialValues={{
          name: customerName || '',
          email: customer.email || '',
          phone: customer.telephone || '',
          deliveryMethod: 'self',
          paymentMethod: 'cash'
        }}
        onSubmit={(values) => handleSubmit(values)}
        validate={validate}
      >
        {({submitForm, values, errors, touched, handleSubmit}) => (
          <div className="checkout-group">
            <div className="order-container">
              <Form className="checkout-form" autoComplete="off">
                <PersonalInfo touched={touched} errors={errors} values={values}/>
                <DeliverMethods/>
                <PaymentMethods/>

                {/* NOT WORKING, DONT RECEIVE VALUE */}
                <Field
                  as={TextField}
                  className="comments"
                  type="text"
                  multiple
                  rows={4}
                  placeholder="Комментарий"
                  variant="outlined"
                  name="comment"
                />
              </Form>
            </div>

            {/* Items In Cart */}
            <ItemsInCart submitBtnFunc={submitForm}/>
            {/*      <SimpleModal onClick={submitForm}/> */}
            {/*     <Button type='button' onClick={sendOrderNAC}>Не авторизированный</Button>
            <Button type='button' onClick={sendOrderAC}>Авторизированный</Button> */}
          </div>
        )}
      </Formik>
    </div>
  );
}

export default CheckoutPage;
