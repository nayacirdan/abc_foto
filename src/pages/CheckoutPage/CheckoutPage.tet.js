import React from 'react';
import { shallow } from 'enzyme';
import CheckoutPage from './CheckoutPage';

describe('test checkout page', () => {
  it('should render checkout page without props', () => {
    const wrapper = shallow(<CheckoutPage />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render checkout page with title', () => {
    const wrapper = shallow(<CheckoutPage />);
    const title = wrapper.find('[data-test-id="checkout-page-title"]');
    expect(title).toBeDefined();
  });
});
