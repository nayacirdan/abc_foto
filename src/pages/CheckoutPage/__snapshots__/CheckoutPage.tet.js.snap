// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test checkout page should render checkout page without props 1`] = `
<div
  className="order-wrapper"
>
  <h2
    className="head-title"
  >
    Оформление заказа
  </h2>
  <Formik
    initialValues={
      Object {
        "email": "",
        "name": "",
        "phone": "",
      }
    }
    onSubmit={[Function]}
    validate={[Function]}
  >
    <Component />
  </Formik>
</div>
`;
